relayer/relays/beacon/header/header.go

```diff
@@ -452,6 +354,11 @@ func (h *Header) syncLaggingSyncCommitteePeriods(ctx context.Context, latestSync
 		}
 	}
 
+	currentSyncPeriod := h.syncer.ComputeSyncPeriodAtSlot(latestSlot)
+	if h.cache.LastSyncedSyncCommitteePeriod < currentSyncPeriod {
+		return ErrSyncCommitteeLatency
+	}
+
```

Comment:
For ErrSyncCommitteeLatency just error exit ASAP and let execution header catch up.